tools/cpp/CMakeLists.txt

```diff
@@ -23,37 +23,6 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 project (generate_geocoding_data)
 
-# Helper functions dealing with finding libraries and programs this library
```

Comment:
As we have moved the download functionality to main library's CMAKE, we were unable to build the offline geocoder individually. Could we not remove this from here OR have this as a common build rule file?

```
CMake Error at CMakeLists.txt:39 (target_link_libraries):
  Target "generate_geocoding_data" links to:

    absl::strings

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```
Note: I am not expertise in CPP builds, please suggest and do what you think is best :)

Reply:
Can we do anything about this..?

Reply:
@penmetsaa, I had assumed (erroneously) that tools/cpp is merely a dependency of the main project, but now I understand you build it independently for other purposes as well.

There are 2, or 3 possible ways to resolve this:

1. Copy the find abseil functions back to tools/cpp. The demerit of this is obviously the maintenance overhead.
2. Create a new shared cmake module for the find abseil functions. This is DRYer than 1, but still a bit of a dance.
3. Build tools/cpp via the main project with an option like `-DBUILD_TOOLS_ONLY=ON`. This should simplify the interface in that all dependency resolutions happen in the same file. It would be my preference too.

I will share a pull request in a moment. Let me know if it works for you, or if you'd rather resolve it a different way.